src/core/src/service_interfaces/LifecycleManagerARC.py

```diff
@@ -0,0 +1,187 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.7+
+
+import json
+import os
+import shutil
+import time
+from core.src.bootstrap.Constants import Constants
+from core.src.service_interfaces.LifecycleManager import LifecycleManager
+
+class LifecycleManagerArc(LifecycleManager):
+    """Class for managing the core code's lifecycle within the extension wrapper"""
+
+    def __init__(self, env_layer, execution_config, composite_logger, telemetry_writer):
+        super(LifecycleManagerArc,self).__init__(env_layer,execution_config,composite_logger,telemetry_writer)
+
+        # Handshake file paths
+        self.ext_state_file_path = os.path.join(self.execution_config.config_folder, Constants.EXT_STATE_FILE)
+        self.core_state_file_path = os.path.join(self.execution_config.config_folder, Constants.CORE_STATE_FILE)
+        #Writing to log
+        self.composite_logger.log_debug("Initializing LifecycleManagerArc")
+        #Variables
+        self.arc_root = "/var/lib/waagent/"
+        self.core_file_name = "CoreState.json"
+        self.arc_extension_folder_pattern = "Microsoft.SoftwareUpdateManagement.LinuxOsUpdateExtension-*"
+        self.config_folder_path = "/config/"
+
+    # region - State checkers
+    def execution_start_check(self):
+        self.composite_logger.log_debug("Execution start check initiating...")
+        extension_sequence = self.read_extension_sequence()
+        core_sequence = self.read_core_sequence()
+        arc_core_sequence = self.read_arc_core_sequence()
+
+        if arc_core_sequence['completed'] == "False":
+            self.composite_logger.log_warning("ARC agent is currently running, sequence number {0}.".format(str(arc_core_sequence['number'])))
+            self.update_core_sequence(completed=True)   # forced-to-complete scenario | extension wrapper will be watching for this event
+            self.env_layer.exit(0)
+
+        if self.execution_config.operation == Constants.AUTO_ASSESSMENT:
+            # newer sequence number has been observed, do not run
+            if int(self.execution_config.sequence_number) < int(extension_sequence['number']) \
+                    or int(self.execution_config.sequence_number) < int(core_sequence['number']):
+                self.composite_logger.log_warning("Auto-assessment not started as newer sequence number detected. [Attempted={0}][DetectedExt={1}][DetectedCore={2}]".format(str(self.execution_config.sequence_number), str(extension_sequence['number']), str(core_sequence['number'])))
+                                                                                                                                                                                                         
+                self.env_layer.exit(0)
+
+
```

Comment:
blank line above exit not required. only one blank line below required. Similar blank-line comments apply to rest of file (see lines 65, 66 for eg)